packages/camera/camera_android_camerax/lib/src/quality_selector.dart

```diff
@@ -0,0 +1,142 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/services.dart';
+import 'package:meta/meta.dart' show immutable;
+
+import 'camera_info.dart';
+import 'camerax_library.g.dart';
+import 'fallback_strategy.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+
+/// Quality setting used to configure components with quality setting
+/// requirements such as creating a Recorder.
+///
+/// See https://developer.android.com/reference/androidx/camera/video/QualitySelector.
+@immutable
+class QualitySelector extends JavaObject {
+  /// Creates a [QualitySelector] with the desired quality and fallback
+  /// strategy, if specified.
+  QualitySelector.from(
+      {BinaryMessenger? binaryMessenger,
+      InstanceManager? instanceManager,
+      required VideoQualityConstraint quality,
+      this.fallbackStrategy})
+      : qualityList = <VideoQualityConstraint>[quality],
+        super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = _QualitySelectorHostApiImpl(
+        binaryMessenger: binaryMessenger, instanceManager: instanceManager);
+    _api.createFromInstance(this, qualityList, fallbackStrategy);
+  }
+
+  /// Creates a [QualitySelector] with ordered desired qualities and fallback
+  /// strategy, if specified.
+  ///
+  /// The final quality will be seleced according to the order in which they are
+  /// specified.
+  QualitySelector.fromOrderedList(
+      {BinaryMessenger? binaryMessenger,
+      InstanceManager? instanceManager,
+      required this.qualityList,
```

Comment:
Do you think it would be good to add a check here that the list isn't empty?

We verify this on the Java side, but the error would end up being the result of code written in dart (i.e. a use of this constructor), so it may be easier to debug if the verification instead comes from the dart constructor.

It would be duplicated code, though, so I'm open to not making this change.

Reply:
Added an `assert` which I think is a happy medium since it will help developers debug.